src/Chapter3.hs

```diff
@@ -479,7 +504,38 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
-
+data Castle = Castle String | NoneCastle
+data Building = Church | Library
+data House = One | Two | Three | Four
+data Wall = Wall | NoWall
+data City = City{
+  cityCastle :: Castle,
+  cityWall :: Wall,
+  cityBuilding :: Building,
+  cityHouses :: [House]
+}
+buildCastle :: City -> String -> City
+buildCastle c n = c {cityCastle = Castle n}
+
+buildHouse :: City -> House -> City
+buildHouse c h = case c of
+  City _ _ _ hs -> c {cityHouses = h:hs}
+
+
+
+amountOfPeople :: [House] -> Int
+amountOfPeople hs = sum (map people hs)
+  where
+    people :: House -> Int
+    people One = 1
+    people Two = 2
+    people Three = 3
+    people Four = 4
+
+buildWalls :: City -> City
+buildWalls c = case (cityCastle c,amountOfPeople (cityHouses c)) of
+  (Castle _, i) -> if(i>=10) then c {cityWall = Wall} else c
```

Comment:
In Haskell functions and operators are always separated by spaces 👾 

```suggestion
  (Castle _, i) -> if i >= 10 then c {cityWall = Wall} else c
```

Also note, that no need to pattern match on the pair, as you will use amount of people only in the first case. 

```haskell
buildWalls c = case cityCastle c of
    Castle _ -> if people >= 10 then c {cityWall = Wall} else c
    _ -> c
  where
     people :: Int
     people = amountOfPeople (cityHouses c)
```